lib/src/main/kotlin/com/swmansion/starknet/service/http/requests/HttpBatchRequestOfDifferentTypes.kt

```diff
@@ -0,0 +1,44 @@
+package com.swmansion.starknet.service.http.requests
+
+import com.swmansion.starknet.data.types.HttpBatchRequestType
+import com.swmansion.starknet.provider.Request
+import com.swmansion.starknet.provider.rpc.JsonRpcRequest
+import com.swmansion.starknet.provider.rpc.buildJsonHttpBatchDeserializerOfDifferentTypes
+import com.swmansion.starknet.service.http.HttpService
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import java.util.concurrent.CompletableFuture
+
+class HttpBatchRequestOfDifferentTypes private constructor(
```

Comment:
I think there should be no distinction between batch reuest and barch request of different types. there should be one class with distinct constructors.

Reply:
Your suggestion is reasonable, but we have to keep in mind that the main difference between these two is the type inference of corresponding function results. `HttpBatchRequest` has a type parameter, while `HttpBatchRequestOfDifferentTypes` does not. Because of that, `batchRequests` results will be fully inferred, while the other ones not. This lets for comfortable usage if we know that we will make requests only of one type.
Does that make more sense to you?

Reply:
I think there should be a way to merge those two, and it would be a lot simpler for users to just be able to rely on one class with multiple constructors

Reply:
`HttpBatchRequestOfDifferentTypes` has been removed in favor of enhancing simplicity. Two constructors have been introduced in `HttpBatchRequest` - one for  single and different types of requests respectively.
It caused a little issue with type erasure in JVM. I've introduced `containsMultipleRequestTypes` boolean flag . It helps to differentiate the overloaded constructors without altering the logic of our code, solving the possible signature clash.